src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -854,6 +854,10 @@ private function registerRouterConfiguration(array $config, ContainerBuilder $co
         $container->setParameter('request_listener.http_port', $config['http_port']);
         $container->setParameter('request_listener.https_port', $config['https_port']);
 
+        $container->setParameter('router.request_context.host', $config['context']['host']);
```

Comment:
`router.request_context.host` parameter value will be `%router.request_context.host%` by default, I don't see how it can work.

Reply:
We need to change the [definition](https://github.com/symfony/symfony/blob/5c37ab016ca71506ad4bfd9f16a5aca8896289c8/src/Symfony/Bundle/FrameworkBundle/Resources/config/routing.xml#L81-L88) arguments instead of setting these parameters from config.
Which means removing those from the xml config (or empty them) and call `replaceArgument` on the definition from here with config values.
Existing `router.request_context.*` will never be used again. They'll only set the defaults for the semantic config.

Reply:
Can we safely stop defining those parameters? What if someone relies on them but don't explicitly set them.

Reply:
Old projects will keep using these parameters, which will provide the default values to the semantic config: no break.
If they have code relying on these parameters elsewhere, well, they already control the parameters values, hence we're fine, no break.

Third parties relying on these params directly looks not legit to me. the contract for these was always to allow the user to set them. Not for third-parties to consume them. They should access the request context instead. No break.

New projects will rely on the semantic config and we could encourage developers to use theses nos instead of the params by creating/updating the recipe. Hence these parameters will naturally become a legacy & internal thing.